tools/careful-cat.c

```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+
+/* Carefully cat stdin to stdout.
+ *
+ * This uses small writes and handles EAGAIN from write by waiting a
+ * bit and trying again.
```

Comment:
Lets reference the 'why' here. Why does this script exist? Links to discussions and bugs would be perfect.